docs/sbft-formal-model/application.s.dfy

```diff
@@ -0,0 +1,63 @@
+//#title Host protocol
+//#desc Define the host state machine here: message type, state machine for executing one
+//#desc host's part of the protocol.
+
+// See exercise01.dfy for an English design of the protocol.
+
+include "network.s.dfy"
+
+module Application {
+  import opened Library
+  type State(!new, ==)
+  type ClientRequest(!new, ==)
+  type ClientReply(!new, ==)
+  
+  function InitialState() : State
+  function Transition(state:State, clientRequest:ClientRequest) : (result:(State, ClientReply))
+  // This is an assumption given to us by the user.
+
+  datatype Variables = Variables(
+    state:State,
+    unprocessedRequests:set<ClientRequest>,
+    undeliveredReplies:set<ClientReply>
+  )
+
+  predicate Init(v:Variables)
+  {
+    && v.s == InitialState()
+  }
+
+  predicate AcceptRequest(v:Variables, v':Variables, clientRequest:ClientRequest)
+  {
+    && v' == v.(unprocessedRequests := v.unprocessedRequests + {clientRequest})
+  }
+
+  predicate ProcessRequest(v:Variables, v':Variables, clientRequest:ClientRequest)
+  {
+    && clientRequest in v.unprocessedRequests
+    && var newState, clientReply := Transition(v.state, clientRequest);
+    && v' == v.(state := newState)
+              .(undeliveredReplies := v.undeliveredReplies + {clientReply})
+              .(unprocessedRequests := v.unprocessedRequests - {clientRequest})
+  }
+
+  predicate DeliverReply(v:Variables, v':Variables, clientReply:ClientReply)
+  {
+    && clientReply in v.undeliveredReplies
+    && v' == v.(undeliveredReplies := v.undeliveredReplies - {clientReply})
+  }
+
+  // JayNF
```

Comment:
Let's be more explicit here: "Jay Normal Form (JNF)". You could also provide a link to an explanation of what it means.